go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -2169,6 +2170,368 @@ func (s *VtctldServer) UpdateCellsAlias(ctx context.Context, req *vtctldatapb.Up
 	}, nil
 }
 
+// Validate is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) Validate(ctx context.Context, req *vtctldatapb.ValidateRequest) (*vtctldatapb.ValidateResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.Validate")
+	defer span.Finish()
+
+	span.Annotate("ping_tablets", req.PingTablets)
+
+	resp := vtctldatapb.ValidateResponse{}
+	getKeyspacesCtx, getKeyspacesCancel := context.WithTimeout(ctx, *topo.RemoteOperationTimeout)
```

Comment:
One change to the implementation is I bounded all the remote calls (both to the topo and to tablet tmc calls) with timeouts. If I didn't do this, I could kill a tablet, and the Validate call would hang until action_timeout, which defaults to 24 hours.